### ISSUE TITLE
USER ISSUE: Previously invisible log caused crash on collection of newly loaded world.

### ISSUE THREAD

**Issue by ParadoxedHope** [StrangeLoopGames/EcoIssues#6568]
**Version:** 0.7.2.5 beta

**Steps to Reproduce:**
I had just loaded up my world, stepped outside to enjoy the view as I drunk my morning cawfee. I noticed a piece of 'tree' where I had cut a tree yesterday (I'm very careful about collecting everything, that piece wasn't there yesterday.) The moment I pressed 'E' to pick it up, the game crashed (crash log below.) 

EDIT: Reloaded world, log still on the ground yet I had been given 3 x Carryable logs. I put those logs in stockpile and attempted to pick up the log again. Identical crash occurred. Reloaded again, 3 more logs in hand, plus the three in my stockpile.

Attempted to change behaviour by digging the ground below the ground log. The log fell into the hole as would be expected (reacting to world physics.) Tried a third time to collect the log, same crash occurred.

I've decided to keep it as a pet now, his name is Bruce, he enjoys long rolls on the beach and jokes about rocks. Not much of a talker but it's nice to have a friend in my single player world until you let us keep pet rabbits. 

Can provide the world save if you would like it for testing purposes :)

**Expected behavior:**
Not crashed?

**Actual behavior:**
The following error message popped up:
```
Caught exception:

<size=60.00%>Caught exception invoking RPC PlayerInteract on Player!

Object reference not set to an instance of an object.


Stack:
   at TreeEntity.PickupLog(Player player, Guid logID)
   at TreeEntity.OnActInteract(InteractionContext context)
   at Eco.Gameplay.Interactions.InteractionExtensions.ExecuteUntilSuccess(IEnumerable`1 results)
   at Eco.Gameplay.Players.Player.PlayerInteract(InteractionInfo info)



Stack:
   at Eco.Shared.Networking.RPCManager.TryInvoke(Object controller, String methodname, BSONObject bsonArgs, Object& result)
   at Eco.Shared.Networking.RPCManager.InvokeOn(BSONObject bson, Object controller, String methodname)
   at Eco.Shared.Networking.RPCManager.HandleReceiveRPC(INetClient client, BSONObject bson)
   at Eco.Shared.Networking.NetPort.ProcessBuffer()
   at Eco.Shared.Networking.NetPort.Update()
   at Eco.Plugins.Networking.Clients.Client.<.ctor>b__56_0()
</size>
```

**Comment by MarjieAVolk** [comment]
If you have a save file where this always happens, we would love to get our hands on that!  Can't wait to meet Bruce, haha.

Duplicate of https://github.com/StrangeLoopGames/EcoIssues/issues/5595

**Comment by ParadoxedHope** [comment]
Bruce has been hopping with excitement to meet you! He actually bounces everytime you update something, like a path, close to him lol

What is the process to get the save file to you? Is it just dragging the .eco and .db files for the save onto here? I tried dragging them to the bottom of this chat box but it says "We don't support that file type."

**Comment by MarjieAVolk** [comment]
XD

I'd recommend zipping the files into a zip folder.  Then you can drop it into the github thread easily.

**Comment by ParadoxedHope** [comment]
The .eco file doesn't seem to be compressing very well, if I zip that file alone it goes from 91.5 Mb to 90.5Mb lol That said, I'm not particularly tech savvy so the problem could exist between user and keyboard. I'll keep trying to get it under 10Mb so I can post it here.

**Comment by MarjieAVolk** [comment]
Well actually the .eco game file is already a compressed file, so it's not surprising it doesn't get much smaller.  It's mostly the .zip file extension that allows github to accept it.

If it is too big, you can try uploading it to dropbox or google drive or some such place.  Also, if you want to only send the .eco file and forget the .db file, that will be fine.

**Comment by ParadoxedHope** [comment]
I was just reading about that, learn something new every day! See if this works;

http://www.filehosting.org/file/details/732373/Bruce.zip

<3

**Comment by MarjieAVolk** [comment]
Got it, thanks!  I will hand it off to my team mate to debug.

**Comment by ParadoxedHope** [comment]
Hmmm, weird, I saved the game, looking at Bruce then shut down Eco to take a break. I've just loaded it up for the first time and Bruce is gone completely :( I'll look around to see if he went for a walk down the hill and let you know if he is still around in that save state.

Sorry about this, he was there when I saved it and had returned after all three reloads following the crashes earlier, so presumed he wood still be there now. I'll let you know if I find him and hopefully at least the crash 'logs' can shed some light on what happened.

**Comment by MarjieAVolk** [comment]
Okay :'(  I was about to ask for help finding him.  I found your house.  I also found a mostly whole felled tree in your garden.  But I assumed Bruce was a small tree slice.

**Comment by ParadoxedHope** [comment]
He was, he was on that ramp going up to the garden.

Interestingly, that log in the garden wasn't there before, it had vanished whilst I was chopping in that area much earlier whilst clearing for the farm. I know that bug has been reported elsewhere and is the same sort of thing that happened with Bruce just reappearing following a reload.

Sorry about this!

**Comment by ParadoxedHope** [comment]
I just chopped that big log, starting at the end deepest into the farm, as I took the third hit and collected, the rest of the log vanished. I'm fairly sure based on the length that there would have been at least 3 pieces left, 6 total as an estimate. Not sure if that is repeatable your end?

**Comment by MarjieAVolk** [comment]
No problem, we have been trying to repro this for months, so it's always worth a shot.

I think I might have found something, is this Bruce?
![image](https://user-images.githubusercontent.com/23460210/38060733-cf38ef00-32a0-11e8-9607-082db17dcb29.png)

**Comment by MarjieAVolk** [comment]
Well, I chopped up and picked up all the logs in that screenshot, without any issues.  Haven't found any more yet.  So I think it might be a no-go unfortunately.  It's possibly one of your backup saves would still have it.  But it's also possible that it's computer specific, or that it's random whether he'll show up every time.  Like I said, we don't know how to repro it yet so, no worries either way.

**Comment by ParadoxedHope** [comment]
Naw that wasn't Bruce, Bruce was down by the house, I had literally built the dirt ramp under him initially to try and roll him into the house so he was warmer at night. It was random that I eventually found good Rice ground up there so it turned into a path. Maybe Bruce was trying to show me the way to the good rice ground? 

Interesting that you chopped up that large log without issue, yet mine disappears. Only other thing that I can think, more specific to how I harvest is that I tend to just hold down 'E' (my collect item keybind), then I'll mine or chop by pressing down RMB. So the moment the resource breaks, E is already held so it collects it. Is it possible that method is creating an issue by thing happening too fast?

**Comment by MarjieAVolk** [comment]
Hmmmmm that is possible.  I can try doing it that way.

**Comment by MarjieAVolk** [comment]
Wow, holding down lmb and e all the time is really nice haha.  But no issues with those logs so far.

**Comment by MarjieAVolk** [comment]
I will note that idea on the other bug though

**Comment by ParadoxedHope** [comment]
"Wow, holding down lmb and e all the time is really nice haha. But no issues with those logs so far."

Haha innit though, so nice mining too as it grabs the little bits and you just scan and then RMB the big ones.

I actually just shut down Eco to grab the small patch. When I loaded back in, I was no longer down in the mine, instead I spawned in the farm, right next to that bit of log that vanished whilst I was chopping, the half that was left had reappeared. I was able to harvest and collect it without issue.

Thanks for taking the time to look at this with me, hope you have a great day!

R.I.P. Bruce, I miss you already little buddy <3
